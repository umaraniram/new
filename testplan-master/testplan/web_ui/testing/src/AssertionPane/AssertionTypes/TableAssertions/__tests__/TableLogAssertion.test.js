import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import TableLogAssertion from '../TableLogAssertion';

function defaultProps() {
  return {
    assertion: {
      "category":"DEFAULT",
      "machine_time":"2019-02-12T17:41:43.241786+00:00",
      "description":"Table Log: list of dicts",
      "line_no":472,
      "display_index":false,
      "meta_type":"entry",
      "columns":["age","name"],
      "indices":[0,1,2],
      "table":[
        {"age":32,"name":"Bob"},
        {"age":24,"name":"Susan"},
        {"age":67,"name":"Rick"}
      ],
      "type":"TableLog",
      "utc_time":"2019-02-12T17:41:43.241777+00:00"
    }
  };
}


describe('FixMatchAssertion', () => {
  let props;
  let shallowComponent;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    shallowComponent = undefined;
  });

  it('shallow renders the correct HTML structure', () => {
    shallowComponent = shallow(<TableLogAssertion {...props}/>);
    expect(shallowComponent).toMatchSnapshot();
  });
});