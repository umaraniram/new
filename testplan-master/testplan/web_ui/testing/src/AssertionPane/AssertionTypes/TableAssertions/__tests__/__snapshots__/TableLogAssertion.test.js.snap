// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixMatchAssertion shallow renders the correct HTML structure 1`] = `
<TableBaseAssertion
  columns={
    Array [
      Object {
        "cellStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
          "fontSize": "small",
          "padding": "6px 11px 6px 11px",
          "width": "72px",
        },
        "field": "id",
        "headerStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
          "width": "72px",
        },
        "title": "ID",
      },
      Object {
        "cellStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
          "fontSize": "small",
          "padding": "6px 11px 6px 11px",
        },
        "field": "age",
        "headerStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
        },
        "title": "age",
      },
      Object {
        "cellStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
          "fontSize": "small",
          "padding": "6px 11px 6px 11px",
        },
        "field": "name",
        "headerStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
        },
        "title": "name",
      },
    ]
  }
  rows={
    Array [
      Object {
        "age": 32,
        "id": 0,
        "name": "Bob",
      },
      Object {
        "age": 24,
        "id": 1,
        "name": "Susan",
      },
      Object {
        "age": 67,
        "id": 2,
        "name": "Rick",
      },
    ]
  }
/>
`;
